server: NPE in checkForUpdates

```
08:56 $ make build && rm -rf cockroach-data/ && ./cockroach start --insecure
build:     alpha.v1-1402-g4fc7dd8 @ 2016/03/25 12:55:59 (go1.6)
admin:     http://macts.local:8080
sql:       postgresql://root@macts.local:26257?sslmode=disable
logs:      cockroach-data/logs
store[0]:  path=cockroach-data
panic: runtime error: invalid memory address or nil pointer dereference
[signal 0xb code=0x1 addr=0x0 pc=0x4168768]

goroutine 75 [running]:
panic(0x5080220, 0xc8200140c0)
	/usr/local/Cellar/go/1.6/libexec/src/runtime/panic.go:464 +0x3e6
github.com/cockroachdb/cockroach/server.(*Server).checkForUpdates(0xc8200ec8c0)
	/Users/tschottdorf/go/src/github.com/cockroachdb/cockroach/server/updates.go:180 +0xc08
github.com/cockroachdb/cockroach/server.(*Server).(github.com/cockroachdb/cockroach/server.checkForUpdates)-fm()
	/Users/tschottdorf/go/src/github.com/cockroachdb/cockroach/server/updates.go:106 +0x20
github.com/cockroachdb/cockroach/server.(*Server).maybeRunPeriodicCheck(0xc8200ec8c0, 0x5232be0, 0xd, 0xc820073f40, 0xf, 0xf, 0xc8208f5e28, 0xece872c79)
	/Users/tschottdorf/go/src/github.com/cockroachdb/cockroach/server/updates.go:153 +0x899
github.com/cockroachdb/cockroach/server.(*Server).maybeCheckForUpdates(0xc8200ec8c0, 0x28a73d5e)
	/Users/tschottdorf/go/src/github.com/cockroachdb/cockroach/server/updates.go:106 +0x75
github.com/cockroachdb/cockroach/server.(*Server).periodicallyCheckForUpdates.func1()
	/Users/tschottdorf/go/src/github.com/cockroachdb/cockroach/server/updates.go:88 +0x59
github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker.func1(0xc82034ea80, 0xc8204dec40)
	/Users/tschottdorf/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:139 +0x52
created by github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker
	/Users/tschottdorf/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:140 +0x62
✘-2 ~/go/src/github.com/cockroachdb/cockroach [master|✔]
08:57 $ git rev-parse HEAD
4fc7dd8318f3722cf33b5eb6e305d62155eafd3e
```

--- a/server/updates.go
+++ b/server/updates.go
@@ -176,6 +176,7 @@ func (s *Server) checkForUpdates() {
 		if log.V(2) {
 			log.Warning("Error checking for updates: ", err)
 		}
+		return
 	}
 	defer res.Body.Close()
 

